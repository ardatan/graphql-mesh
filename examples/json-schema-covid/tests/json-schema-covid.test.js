const { findAndParseConfig } = require('@graphql-mesh/config');
const { getMesh } = require('@graphql-mesh/runtime');
const { basename, join } = require('path');

const { introspectionFromSchema, lexicographicSortSchema } = require('graphql');
const { readFile } = require('fs-extra');

const config$ = findAndParseConfig({
  dir: join(__dirname, '..'),
});
const mesh$ = config$.then(config => getMesh(config));
jest.setTimeout(30000);

describe('JSON Schema Covid', () => {
  it('should generate correct schema', async () => {
    const { schema } = await mesh$;
    expect(
      introspectionFromSchema(lexicographicSortSchema(schema), {
        descriptions: false,
      })
    ).toMatchSnapshot('json-schema-covid-schema');
  });
  it('should give correct response for STEP 1: 2 sources side by side', async () => {
      const getDataStep1Query = await readFile(join(__dirname, '../example-queries/getData_step1.graphql'), 'utf8');
      const { execute } = await mesh$
      const result = await execute(getDataStep1Query);
      expect(typeof result?.data?.case?.confirmed).toBe('number');
      expect(result?.data?.case?.countryRegion).toBe('France');
      expect(typeof result?.data?.case?.deaths).toBe('number');
      expect(typeof result?.data?.case?.recovered).toBe('number');

      expect(result?.data?.population?.records).toHaveLength(1);
      expect(result?.data?.population?.records[0]?.fields?.country_name).toBe('France');
      expect(typeof result?.data?.population?.records[0]?.fields?.value).toBe('number');
  });
  it('should give correct response for STEP1: 2 sources side by side', async () => {
    const getDataStep1Query = await readFile(join(__dirname, '../example-queries/getData_step1.graphql'), 'utf8');
    const { execute } = await mesh$;
    const result = await execute(getDataStep1Query);
    expect(typeof result?.data?.case?.confirmed).toBe('number');
    expect(result?.data?.case?.countryRegion).toBe('France');
    expect(typeof result?.data?.case?.deaths).toBe('number');
    expect(typeof result?.data?.case?.recovered).toBe('number');

    expect(result?.data?.population?.records?.length).toBe(1);
    expect(result?.data?.population?.records[0]?.fields?.country_name).toBe('France');
    expect(typeof result?.data?.population?.records[0]?.fields?.value).toBe('number');
  });
  it('should give correct response for STEP2: 2 sources combined', async () => {
    const getDataStep2Query = await readFile(join(__dirname, '../example-queries/getData_step2.graphql'), 'utf8');
    const { execute } = await mesh$;
    const result = await execute(getDataStep2Query);
    expect(typeof result?.data?.case?.confirmed).toBe('number');
    expect(typeof result?.data?.case?.deaths).toBe('number');
    expect(typeof result?.data?.case?.recovered).toBe('number');

    expect(result?.data?.case?.population?.records?.length).toBe(1);
    expect(typeof result?.data?.case?.population?.records[0]?.fields?.value).toBe('number');
  });
  it('should give correct response for STEP3_1: 2 sources combined to get ratios', async () => {
    const getDataStep3_1Query = await readFile(join(__dirname, '../example-queries/getData_step3_1.graphql'), 'utf8');
    const { execute } = await mesh$;
    const result = await execute(getDataStep3_1Query);
    expect(typeof result?.data?.fr?.deathRatio).toBe('number');

    expect(typeof result?.data?.at?.deathRatio).toBe('number');
  });
  it('should give correct response for STEP3_2: 2 sources combined to get ratios & case & population', async () => {
    const getDataStep3_2Query = await readFile(join(__dirname, '../example-queries/getData_step3_2.graphql'), 'utf8');
    const { execute } = await mesh$;
    const result = await execute(getDataStep3_2Query);
    expect(typeof result?.data?.fr?.deathRatio).toBe('number');
    expect(typeof result?.data?.fr?.case?.deaths).toBe('number');
    expect(result?.data?.fr?.population?.records?.length).toBe(1);
    expect(typeof result?.data?.fr?.population?.records[0]?.fields?.value).toBe('number');

    expect(typeof result?.data?.at?.deathRatio).toBe('number');
    expect(typeof result?.data?.at?.case?.deaths).toBe('number');
    expect(result?.data?.at?.population?.records?.length).toBe(1);
    expect(typeof result?.data?.at?.population?.records[0]?.fields?.value).toBe('number');
  });
  afterAll(() => mesh$.then(mesh => mesh.destroy()));
});
