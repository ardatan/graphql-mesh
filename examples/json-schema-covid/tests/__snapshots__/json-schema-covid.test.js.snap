// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON Schema Covid should generate correct schema: json-schema-covid-schema 1`] = `
"\\"\\"\\"Desc Api Population\\"\\"\\"
type ApiPopulation {
  records: [Result]
}

\\"\\"\\"\\"\\"\\"
enum CacheControlScope {
  \\"\\"\\"\\"\\"\\"
  PRIVATE

  \\"\\"\\"\\"\\"\\"
  PUBLIC
}

\\"\\"\\"\\"\\"\\"
type Case {
  \\"\\"\\"\\"\\"\\"
  active: Int

  \\"\\"\\"\\"\\"\\"
  admin2: Int

  \\"\\"\\"\\"\\"\\"
  combinedKey: Int

  \\"\\"\\"\\"\\"\\"
  confirmed: Int

  \\"\\"\\"\\"\\"\\"
  countryRegion: String

  \\"\\"\\"\\"\\"\\"
  deaths: Int

  \\"\\"\\"\\"\\"\\"
  fips: Int

  \\"\\"\\"\\"\\"\\"
  iso2: String

  \\"\\"\\"\\"\\"\\"
  iso3: String

  \\"\\"\\"\\"\\"\\"
  lastUpdate: Date

  \\"\\"\\"\\"\\"\\"
  lat: Float

  \\"\\"\\"\\"\\"\\"
  long: Float
  population: ApiPopulation

  \\"\\"\\"\\"\\"\\"
  provinceState: String

  \\"\\"\\"\\"\\"\\"
  recovered: Int
}

\\"\\"\\"\\"\\"\\"
type Count {
  \\"\\"\\"\\"\\"\\"
  confirmed: Int

  \\"\\"\\"\\"\\"\\"
  deaths: Int

  \\"\\"\\"\\"\\"\\"
  recovered: Int
}

\\"\\"\\"\\"\\"\\"
type Daily {
  \\"\\"\\"\\"\\"\\"
  deltaConfirmed: Int

  \\"\\"\\"\\"\\"\\"
  deltaRecovered: Int

  \\"\\"\\"\\"\\"\\"
  mainlandChina: Int

  \\"\\"\\"\\"\\"\\"
  objectid: Int

  \\"\\"\\"\\"\\"\\"
  otherLocations: Int

  \\"\\"\\"\\"\\"\\"
  reportDate: Date

  \\"\\"\\"\\"\\"\\"
  reportDateString: String

  \\"\\"\\"\\"\\"\\"
  totalConfirmed: Int

  \\"\\"\\"\\"\\"\\"
  totalRecovered: Int
}

\\"\\"\\"\\"\\"\\"
scalar Date

\\"\\"\\"\\"\\"\\"
type Query {
  \\"\\"\\"\\"\\"\\"
  case(countryRegion: String): Case

  \\"\\"\\"\\"\\"\\"
  cases: [Case]

  \\"\\"\\"\\"\\"\\"
  count: Count

  \\"\\"\\"\\"\\"\\"
  daily: [Daily]

  \\"\\"\\"
  GET /?dataset=world-population&q={args.country}&rows=1&sort=year&facet=year&facet=country_name
  \\"\\"\\"
  population(country: String): ApiPopulation
  stat(country: String!): Stat
}

\\"\\"\\"Result of API\\"\\"\\"
type Result {
  fields: _records_items_fields
}

type Stat {
  case: Case
  confirmedRatio: Float
  deathRatio: Float
  population: ApiPopulation
  recoveredRatio: Float
}

\\"\\"\\"The \`Upload\` scalar type represents a file upload.\\"\\"\\"
scalar Upload

type _records_items_fields {
  country_name: String
  value: Float
  year: String
}
"
`;
